### ISSUE TITLE
use conda to install etabackend

### ISSUE THREAD

**Issue by linzuzeng** [timetag/ETA#151]
I just found that one can create a 'pip:' section in environment.yml for conda.

Maybe we can add an environment.yml to ETA, for easier installation with `conda env create -f environment.yml`

Here is an example for the environment.yml

```
name: ETA
channels:
  - conda-forge
  - defaults
dependencies:
  - python=3.8.1=h5fd99cc_1
  - xxxxx
  - pip:
    - etabackend==0.7.xxxx
```

**Comment by Cyberfly100** [comment]
yes, mine looks like this:

name: ETA
channels:
  - conda-forge
dependencies:
  - python=3
  - pip
  - pip:
    - etabackend

not entierly sure if it is necessary to ask for pip the exist before specifying what should be installed via pip

**Comment by linzuzeng** [comment]
Don't you install matplotlib and other libs from conda first? I thought installing matplotlib with etabackend  from pip would break the conda environment.

**Comment by linzuzeng** [comment]
It works though. Close for now

**Comment by linzuzeng** [comment]
The installing script is in 

https://github.com/timetag/ETA/blob/master/gui/upgradebackend.bat

and the environment checking logic is in

https://github.com/timetag/ETA/blob/master/gui/src/main/backend.js

Pay attention to `detached: false` and `detached: true, shell: true`

**Comment by gyger** [comment]
I changed the conda file (bc593ae95e52fbad14650a055e2eba65d1bf0eef) to use mostly conda to install the packages, we only use pip for etabackend (that is the suggested way). [Best practices for mixing pip and conda.](https://www.anaconda.com/blog/using-pip-in-a-conda-environment).

**Comment by linzuzeng** [comment]
Great job! Shall we use conda as default way to install ETA? Or we keep the pip-way and put the conda-way in the documentation.

related installation code in  https://github.com/timetag/ETA/blob/bc593ae95e52fbad14650a055e2eba65d1bf0eef/gui/upgradebackend.bat#L1

**Comment by linzuzeng** [comment]
Shall we zip the conda venv folder (for Windows) so that people with poor or no Internet connection could run ETA with the bundle?

Also it helps sloving the mess created by having multiple offical python installations on the same computer.

**Comment by gyger** [comment]
Yeah, this tool [https://conda.github.io/conda-pack/](conda-pack) could be a good way of distributing the environment around the installer.

Second, I recently discovered Jupyter-desktop, they seem to handle a lot around all this mess with electron and Python, perhaps we could somehow benefit from them.
https://github.com/jupyterlab/jupyterlab-desktop

**Comment by linzuzeng** [comment]
Conda-Pack Seems to be a drop-in replacement at the moment.  

jupyterlab-desktop has Extension support, which could be used for making the GUI. but it's a bit complicated. The windows package is intended for users new to Python. If they already know Jupyter , they must know how to install ETA via pip themselves and they probably already have a jupyter installed.

I agree that using Jupyter as a replacement of the existing display panel could be a very interesting direction to go. We still ne to figure out how the jupyter GUI and ETA GUI could be combined.

**Comment by linzuzeng** [comment]
Fixed via 1af53711f33d90698f12e0609bb75dbf11d8b0d3
